data/adapters/file_fileset_adapter.py

```diff
@@ -311,43 +356,65 @@ def query_fileset_files_props_encode(self, accession):
 
         preferred_assay_titles = set()
         assay_term_ids = set()
-        prediction = False
-        prediction_method = None
+        method = None
+        class_type = None
         if dataset_type == 'Annotation':
-            prediction, prediction_method = self.parse_annotation(
-                dataset_object, portal_url, prediction, prediction_method, software, preferred_assay_titles, assay_term_ids)
+            class_type, method = self.parse_annotation(
```

Comment:
it's a bit confusing that only `class_type` and `method` in return while the function also modifies some other parameters like `preferred_assay_titles`, but it works fine.

Reply:
I can have it return the rest of the properties.